### PR TITLE
Add new consumer method for single topic with retryEnabled flag

### PR DESCRIPTION
**Pull request by sagarrohankar-bsft**

### CODE REVIEW

File: pulsarlib/messaging.go [hunk 1]
```diff
@@ -392,6 +392,75 @@ func CreateConsumer(tenantID string, namespace string, topics []string, subscrip
...
+	The Unpause() function will resume receiving messages.
+	The Stop() function will flush existing messages and stop the consumer. It won't delete the subscription.
+	The Unsubscribe() function can be used if subscription needs to be deleted.
+	The Stats() function provides the stats for messages consumed.
+
+	Creating multiple instances of Consumer for same topic will deliver message to only one of the instances.
+	Inorder to recreate a Consumer for same topic make sure Stop() is called on old Consumer instance.
+    retryEnabled will let the consumer retry message in case of HandleMessage return `RetryMessage` struct.
+*/
+func CreateSingleTopicConsumer(tenantID string, namespace string, topic string, subscriptionName string, handler Handler, commitInterval uint64, retryEnabled bool) (Consumer, error) {
```
Comment (merajblueshift) [inline pulsarlib/messaging.go:411]: Let us deprecate commitInterval

File: pulsarlib/messaging.go [hunk 2]
```diff
@@ -392,6 +392,75 @@ func CreateConsumer(tenantID string, namespace string, topics []string, subscrip
...
+	if msging == nil {
+		return nil, fmt.Errorf("InitMessaging not called yet")
+	}
+
+	topicURI := fmt.Sprintf("persistent://%s/%s/%s", tenantID, namespace, topic)
+
+	consumerOptions := pulsar.ConsumerOptions{
+		Topic:            topicURI,
+		SubscriptionName: subscriptionName,
+		Type:             pulsar.Shared,
```
Comment (merajblueshift) [inline pulsarlib/messaging.go:422]: Let us add this option to consumer creation
`SubscriptionInitialPosition: pulsar.SubscriptionPositionEarliest,`

Comment (merajblueshift) [reply pulsarlib/messaging.go:422]: We can use Earliest as default and accept an option to override it. 
In order to keep function clean let us accept subscriptionName, retryEnabled, iniitialPosition via a struct called ConsumerOptions

File: pulsarlib/messaging.go (hunk 1 above)
Comment (sagarrohankar-bsft) [reply pulsarlib/messaging.go:411]: then when should we commit message?

Comment (merajblueshift) [reply pulsarlib/messaging.go:411]: Our commit on consumer is currently a NoOp. We can remove it altogether.

Comment (sagarrohankar-bsft) [reply pulsarlib/messaging.go:411]: got it

File: pulsarlib/messaging.go [hunk 3]
```diff
@@ -361,24 +379,93 @@ func CreateConsumer(tenantID string, namespace string, topics []string, subscrip
 	for _, tp := range topics {
 		topicArr = append(topicArr, fmt.Sprintf("persistent://%s/%s/%s", tenantID, namespace, tp))
 	}
-	c, err := msging.client.Subscribe(pulsar.ConsumerOptions{
+	consumerOptions := pulsar.ConsumerOptions{
```
Comment (merajblueshift) [inline pulsarlib/messaging.go:382]: Can we put validation on opts passed. Subscription name can be empty. Does pulsar raise an error in that case

Comment (sagarrohankar-bsft) [reply pulsarlib/messaging.go:382]: yes, it raise an error if subscription is empty

**Pull request merged**